pydevices/HtsDevices/acq2106_WRPG.py

```diff
@@ -0,0 +1,193 @@
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+import MDSplus
+import time
+import numpy
+import os
+from tempfile import mkstemp
+
+class ACQ2106_WRPG(MDSplus.Device):
+    """
+    D-Tacq ACQ2106 with ACQ423 Digitizers (up to 6)  real time streaming support.
+
+    DIO with 32 channels
+
+    MDSplus.Device.debug - Controlled by environment variable DEBUG_DEVICES
+		MDSplus.Device.dprint(debuglevel, fmt, args)
+         - print if debuglevel >= MDSplus.Device.debug
+
+    """
+
+    parts=[
+        {'path':':NODE',        'type':'text',                     'options':('no_write_shot',)},
+        {'path':':COMMENT',     'type':'text',                     'options':('no_write_shot',)},
+        {'path':':TRIG_TIME',   'type':'numeric',                  'options':('write_shot',)},
+        {'path':':RUNNING',     'type':'numeric',                  'options':('no_write_model',)},
+        {'path':':LOG_OUTPUT',  'type':'text',   'options':('no_write_model', 'write_once', 'write_shot',)},
+        {'path':':INIT_ACTION', 'type':'action', 'valueExpr':"Action(Dispatch('CAMAC_SERVER','INIT',50,None),Method(None,'INIT',head))",'options':('no_write_shot',)},
+        {'path':':STOP_ACTION', 'type':'action', 'valueExpr':"Action(Dispatch('CAMAC_SERVER','STORE',50,None),Method(None,'STOP',head))",      'options':('no_write_shot',)},
+        # {'path':':STL_FILE',    'type':'TEXT'},
+    ]
+
+
+    for j in range(32):
+        parts.append({'path':':OUTPUT_%3.3d' % (j+1,), 'type':'NUMERIC', 'options':('no_write_shot',)})
+
+    def init(self):
+        import acq400_hapi
+        uut = acq400_hapi.Acq400(self.node.data(), monitor=False)
+        
+        #Number of channels of the DIO482
+        nchans = 32
+
+        # uut.s0.SIG_SRC_TRG_0 ='WRTT0'
+        # Setting the trigger in the GPG module. These settings depends very much on what is the
+        # configuration of the experiment. For example, when using one WRTT timing highway, the we can use d0, which will be
+        # the same used by the digitazer module. Otherwise, we can choose a different one, to be a independent highway from
+        # the digitazer, like d1.
+        uut.s0.SIG_EVENT_SRC_0 = 'GPG'
+        uut.s0.GPG_ENABLE    ='enable'
+        uut.s0.GPG_TRG       ='1'    #external=1, internal=0
```

Comment:
Are these supposed to be configurable by changing nodes in the device?

Reply:
Are you thinking that nodes of the device should have the values used to configure the knobs of the ACQ? That can be done, of course.

Reply:
Ok, I have added a new tree node so that we can choose in the model between d0 or d1 for the triggering signal.